unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuildModeHUD/Scripts/Common/SceneCatalogView.cs

```diff
@@ -0,0 +1,155 @@
+    [SerializeField] internal RectTransform categoryRT;
+
+    [Header("MinSize Catalog RectTransforms")]
+    [SerializeField] internal RectTransform panelMinSizeRT;
+    [SerializeField] internal RectTransform headerMinSizeRT;
+    [SerializeField] internal RectTransform searchBarMinSizeRT;
+    [SerializeField] internal RectTransform assetPackMinSizeRT;
+
+    [Header("MaxSize Catalog RectTransforms")]
+    [SerializeField] internal RectTransform panelMaxSizeRT;
+    [SerializeField] internal RectTransform headerMaxSizeRT;
+    [SerializeField] internal RectTransform searchBarMaxSizeRT;
+    [SerializeField] internal RectTransform assetPackMaxSizeRT;
+
+    internal bool isCatalogExpanded = false;
+
+    private const string VIEW_PATH = "Common/SceneCatalogView";
+
+    internal static SceneCatalogView Create()
+    {
+        var view = Instantiate(Resources.Load<GameObject>(VIEW_PATH)).GetComponent<SceneCatalogView>();
+        view.gameObject.name = "_SceneCatalogView";
+
+        return view;
+    }
+
+    private void Awake()
+    {
+        hideCatalogBtn.onClick.AddListener(OnHideCatalogClick);
+        backgBtn.onClick.AddListener(Back);
+        toggleCatalogBtn.onClick.AddListener(ToggleCatalogExpanse);
+    }
+
+    private void OnDestroy()
+    {
+        hideCatalogBtn.onClick.RemoveListener(OnHideCatalogClick);
+        backgBtn.onClick.RemoveListener(Back);
+        toggleCatalogBtn.onClick.RemoveListener(ToggleCatalogExpanse);
+    }
+
+    public void ToggleCatalogExpanse()
+    {
+        if (isCatalogExpanded)
+        {
+            BuilderInWorldUtils.CopyRectTransform(panelRT, panelMinSizeRT);
+            BuilderInWorldUtils.CopyRectTransform(headerRT, headerMinSizeRT);
+            BuilderInWorldUtils.CopyRectTransform(searchBarRT, searchBarMinSizeRT);
+            BuilderInWorldUtils.CopyRectTransform(assetPackRT, assetPackMinSizeRT);
+            BuilderInWorldUtils.CopyRectTransform(categoryRT, assetPackMinSizeRT);
+        }
+        else
+        {
+            BuilderInWorldUtils.CopyRectTransform(panelRT, panelMaxSizeRT);
+            BuilderInWorldUtils.CopyRectTransform(headerRT, headerMaxSizeRT);
+            BuilderInWorldUtils.CopyRectTransform(searchBarRT, searchBarMaxSizeRT);
+            BuilderInWorldUtils.CopyRectTransform(assetPackRT, assetPackMaxSizeRT);
+            BuilderInWorldUtils.CopyRectTransform(categoryRT, assetPackMaxSizeRT);
+        }
+
+        isCatalogExpanded = !isCatalogExpanded;
+    }
+
+    public void OnHideCatalogClick()
+    {
+        OnHideCatalogClicked?.Invoke();
+    }
+
+    public void Back()
+    {
+        OnSceneCatalogBack?.Invoke();
+    }
+
+    public void SetCatalogTitle(string text)
+    {
+        catalogTitleTxt.text = text;
+    }
+
+    public bool IsCatalogOpen()
+    {
+        return gameObject.activeSelf;
+    }
+
+    public void CloseCatalog()
+    {
+        if (gameObject.activeSelf)
+            StartCoroutine(CloseCatalogAfterOneFrame());
```

Comment:
I know this is my code, so my fault! This can be handler with our Courutine handler?

Reply:
Done in https://github.com/decentraland/explorer/pull/2103/commits/30174fad9d952af1fa8930a3c4fbd5dc800fde67 Thanks!